src/StakeVault.sol

```diff
@@ -163,4 +164,24 @@ contract StakeVault is Ownable {
     function amountStaked() public view returns (uint256) {
         return stakeManager.getStakedBalance(address(this));
     }
+
+    /**
+     * @notice Allows vaults to exit the system in case of emergency or the system is rigged.
+     * @param _destination The address to receive the funds.
+     * @dev This function tries to read `IStakeManager.emergencyModeEnabeled()` to check if an
+     *      emergency mode is enabled. If the call fails, it will still transfer the funds to the
+     *      destination address.
+     * @dev This function is only callable by the owner.
+     * @dev Reverts when `emergencyModeEnabled()` returns false.
+     */
+    function emergencyExit(address _destination) external onlyOwner validDestination(_destination) {
+        try stakeManager.emergencyModeEnabled() returns (bool enabled) {
+            if (!enabled) {
+                revert StakeVault__NotAllowedToExit();
+            }
+            STAKING_TOKEN.transfer(_destination, STAKING_TOKEN.balanceOf(address(this)));
+        } catch {
+            STAKING_TOKEN.transfer(_destination, STAKING_TOKEN.balanceOf(address(this)));
+        }
+    }
```

Comment:
So here, in theory, if there was a malicious upgrade that would cause `emergencyModeEnabled()` to always always revert, funds will not be moved.

This will be fixed in the follow up PR that introduces `leave()` which will **always** move the funds **if** the `StakeManager` is not trusted (which is the case when tehre was a (malicious) upgrade